subsys/debug/coredump/coredump_core.c

```diff
@@ -70,10 +77,36 @@ static void dump_header(unsigned int reason)
 	backend_api->buffer_output((uint8_t *)&hdr, sizeof(hdr));
 }
 
-#if defined(CONFIG_DEBUG_COREDUMP_MEMORY_DUMP_MIN) || \
+#if defined(CONFIG_DEBUG_COREDUMP_MEMORY_DUMP_MIN) ||                                              \
 	defined(CONFIG_DEBUG_COREDUMP_MEMORY_DUMP_THREADS)
+
+static inline void select_stack_region(const struct k_thread *thread, uintptr_t *start,
+				       uintptr_t *end)
+{
+	uintptr_t sp;
+
+	*start = thread->stack_info.start;
+	*end = thread->stack_info.start + thread->stack_info.size;
+
+	if (!IS_ENABLED(CONFIG_DEBUG_COREDUMP_THREAD_STACK_TOP)) {
+		return;
+	}
+
+	sp = arch_coredump_stack_ptr_get(thread);
+
+	if (!IN_RANGE(sp, *start, *end)) {
```

Comment:
I think this logic should be adjusted, only set *start to sp when its in the appropriate range, otherwise leave *start set as thread->stack_info.start. Then we still dump a portion of the stack (or all depending on CONFIGs) even for cases where the sp is some unexpected value.

Reply:
My line of reasoning was that if SP is messed up, dumping an arbitrary portion of stack won't give you much anyway. But after second thought, in some cases (stack overflow) dumping the top of a stack might give a hint what overflew the stack, so I can change this.

Reply:
Yeah, I think stack overflow is a good example

Reply:
ok, done